Feature hours per month counter

maciejSamerdak opened the pull request:
Introduces total hours per month counter for all report list views.

Related #138

rwrzesien commented on employees/views.py:
```diff
@@ -62,10 +62,22 @@ def hours_per_day_counter(reports: list) -> Decimal:
     return hours_sum
 
 
+def hours_per_month_counter(dictionary: dict) -> Decimal:
```
I think this could be fully achieved in queryset.

rwrzesien commented on employees/views.py:
```diff
@@ -219,7 +235,7 @@ def include_users_in_reports_dict(self, project: Project, year: int, month: int)
             user_reports_dict = query_as_dict(queryset)
             key = user.email
-            reports_dict[key] = user_reports_dict
+            reports_dict[key] = [user_reports_dict, decimal_to_hours_string(hours_per_month_counter(user_reports_dict))]
```
This is fundamentally wrong because there should be no need to iterate through members and calculate hours for each separately, it should be also done in pull request, but lets not over complicate it now unless you want to refactor the whole method?

rwrzesien commented:
@maciejSamerdak 
Please try to apply the same approach as in https://github.com/Code-Poets/sheetstorm/pull/137#issuecomment-491627064

If you can't do this or don't have time please let me know, I can do this but wanted to leave it to you to see how it can be simplified.

rwrzesien commented on employees/views.py:
```diff
@@ -55,6 +55,14 @@ def daily_hours_sum(queryset: QuerySet) -> dict:
     )
 
 
+def monthly_hours_sum(queryset: QuerySet) -> dict:
```
Please also make this a `ReportManager` method, but I would like to understand it correctly - in this queryset you want to sum of all work hours for each author for each month? Because what it currently does is getting sum of all work hours for each author but from the whole queryset - if queryset contains data from more then a month, it is calculated for that whole time period, without dividing into months, only by author.

rwrzesien replied:
@maciejSamerdak What about that ^ ?

maciejSamerdak replied:
Correct, this issue would be resolved with next PRs, which add date filtering in list views showing only one month on page. Having that in mind I simplified this function, since the queryset provided in the view would already be filtered. Again, this feature is **fast-forwarded**, but @Karrp wanted it introduced ASAP, knowing this issue.

rwrzesien commented:
@maciejSamerdak Same here, please make it a `ReportManager` method and lets clarify what the queryset should do exactly.

maciejSamerdak commented:
I've moved hours counting function to **ReportManager** in same fashion as in #137.

There's still one issue. Currently this feature is also introduced to the regular **report list**...
But it won't work correctly, until #139 is merged. Currently, this sums up work hours from all reports displayed on page, so it will work properly as soon as month segregation is introduced.

Should I keep it in case #139 is merged first? Should I delete it otherwise?
If it should stay this way, should I also introduce this feature to **author report list**? This one doesn't even have segregation feature in development yet.

rwrzesien commented:
> There's still one issue. Currently this feature is also introduced to the regular report list...
But it won't work correctly, until #139 is merged. Currently, this sums up work hours from all reports displayed on page, so it will work properly as soon as month segregation is introduced.

Lets wait until everything else what is needed is merged.

maciejSamerdak commented:
@Karrp wanted this to be introduced with 0.3.0. I've changed the base to master already.

rwrzesien commented:
@maciejSamerdak Please add test proving that `monthly_hours_sum` method will sum hours grouped by months.

maciejSamerdak commented:
This feature does indeed count hours for entirety of queryset rather than just one month. This is because it was originally meant to be introduced after #138 where views pass querysets that are already filtered. Knowing this issue @Karrp wants it introduced with 0.3.0 due to popular demand.

On @kbeker request, I described it in an issue #239 After each list view is split per month (currently 2 in #138 and #139), we can consider it resolved.

kbeker commented:
After talk with @rwrzesien I'm dismissing his `request changes` from this PR. Everything explained above
